### PR TITLE
Host database

### PR DESCRIPTION
**Pull request by nolixul**
- We are plugged into vercel
- You need to ask me for the secrets to put in your local .env file - I'll share them with you via lastpass or something
- I altered the seeding script to match the sample provided in their tutorial which works with their PSQL SDK, but luckily most of what we already had was usable :)
- Right now we can drop all the tables from vercel (in the website) (or I'm sure there's a cli)
- We can run the seeding script locally, which points to the placeholder data we have
- When we're ready to start using real data, we could theoretically drop all the tables and create new ones in the platform or seed new, empty tables locally and just not populate them

Screenshot 2024-01-19 at 16 25 58

### CODE REVIEW

File: frontend/lib/placeholder-data.js
Comment (321jellyfish) [inline frontend/lib/placeholder-data.js:1]: In the placeholder data why is the an id field with true or false? Shouldn't this be a book_id with a number?

Also the requested field is on the Comment type definition but in the placeholder data it is on Book? This is also inconsistent in the seeding script. 

Could: rename to cover_image_url as we are saving a URL and calling it cover_image implies we are saving an image? Unless we are going to store an actual image in which case can we change the placeholder data? and change the data type to an image?

**Review by 321jellyfish — CHANGES_REQUESTED** [review]
Could: suggest renaming lib folder as I don't know what this stands for. 

Question: should the seeding script be in the front end? Seems more natural to be in the database folder? 

Why do we have two seeds files in the db folder? Do we also need mock and placeholder data? 
I don't understand why there is some database stuff in frontend and some in db - could you explain? Or remove if any is unnecessary.

File: frontend/lib/placeholder-data.js
Comment (nolixul) [reply frontend/lib/placeholder-data.js:1]: > In the placeholder data why is the an id field with true or false? Shouldn't this be a book_id with a number?
> 
> Also the requested field is on the Comment type definition but in the placeholder data it is on Book? This is also inconsistent in the seeding script.
> 
> Could: rename to cover_image_url as we are saving a URL and calling it cover_image implies we are saving an image? Unless we are going to store an actual image in which case can we change the placeholder data? and change the data type to an image?

- id field is an error - I must have messed up when generating the data, I'll re-do it - nice catch :) 
- We can make it book_id, is there a benefit to doing that over id? I forget how tables in databases work
- I'll change it to be on the book type definition, I think this makes more sense!

- I'll leave it as cover_image for now, until we know how we're going to be saving the images. That's kind of up in the air right? And then we can be specific when we get there. It's placeholder data so we can alter this when we start using this for real, unless it's better to be specific now and then change it later. What do you think?

Comment (nolixul) [reply frontend/lib/placeholder-data.js:1]: I see! I've put book_id in type definitions - I'll change it to that when I regenerate the data. Question still stands though if you know the answer?

**Comment by nolixul** [comment]
> Could: suggest renaming lib folder as I don't know what this stands for.
> 
> Question: should the seeding script be in the front end? Seems more natural to be in the database folder?
> 
> Why do we have two seeds files in the db folder? Do we also need mock and placeholder data? I don't understand why there is some database stuff in frontend and some in db - could you explain? Or remove if any is unnecessary.

- The lib (library I assume) I created following the tutorial https://nextjs.org/learn/dashboard-app/getting-started (there's a little explainer on that link) - the placeholder data will be deleted from that folder later.
- We don't need mock data or the db anymore, that was merged in before we decided to use next to host the db :) we can go ahead and delete it now! It was a really helpful reference while writing this, I basically ripped out the seed script to re-use in the frontend seed script. Thanks for reminding me, I'll clean it up

File: frontend/lib/placeholder-data.js
Comment (321jellyfish) [reply frontend/lib/placeholder-data.js:1]: I don't know whether it's better to call it id or book id. 

Could you format the placeholder data so it's human readable? If you right click and format document? 
I don't think that should cause any problems and it's easier to read then.

I would change to cover_image_url as it is currently a url. Or get rid of it entirely if we don't know what we are doing with images and instead we should figure out what we are doing.

Comment (nolixul) [reply frontend/lib/placeholder-data.js:1]: Changes made :)

**Review by 321jellyfish — APPROVED** [review]

**Review by 321jellyfish — APPROVED** [review]

**Review by 321jellyfish — APPROVED** [review]

**Pull request merged**